gui/src/renderer/components/Switch.tsx

```diff
@@ -156,7 +163,14 @@ export default class Switch extends React.PureComponent<IProps, IState> {
     }
 
     if (this.props.isOn !== this.state.isOn) {
-      this.notify();
+      if (
+        this.props.confirmation === undefined ||
+        (await this.props.confirmation(this.state.isOn))
+      ) {
+        this.notify();
+      } else {
+        this.setState({ isOn: !this.state.isOn });
```

Comment:
If I may, updating state based on the current state should be done with a function.

```suggestion
        this.setState((prevState) => ({ isOn: !prevState.isOn }));
```

Reply:
Thanks for your feedback. This has been changed to `this.props.isOn` so this is no longer the case.